pkg/lockfile/lockfile_windows.go

```diff
@@ -102,51 +52,47 @@ func (l *LockFile) GetLastWrite() (LastWrite, error) {
 //
 // The caller must hold the lock for writing.
 func (l *LockFile) RecordWrite() (LastWrite, error) {
-	return LastWrite{}, nil
+	l.AssertLockedForWriting()
+	lw := newLastWrite()
+	lockContents := lw.serialize()
+	ol := new(windows.Overlapped)
+	var n uint32
+	err := windows.WriteFile(windows.Handle(l.fd), lockContents, &n, ol)
+	if err != nil {
+		return LastWrite{}, err
+	}
+	if int(n) != len(lockContents) {
+		return LastWrite{}, windows.ERROR_DISK_FULL
+	}
+	return lw, nil
 }
 
-// ModifiedSince checks if the lock has been changed since a provided LastWrite value,
-// and returns the one to record instead.
-//
-// If ModifiedSince reports no modification, the previous LastWrite value
-// is still valid and can continue to be used.
-//
-// If this function fails, the LastWriter value of the lock is indeterminate;
-// the caller should fail and keep using the previously-recorded LastWrite value,
-// so that it continues failing until the situation is resolved. Similarly,
-// it should only update the recorded LastWrite value after processing the update:
-//
-//	lw2, modified, err := state.lock.ModifiedSince(state.lastWrite)
-//	if err != nil { /* fail */ }
-//	state.lastWrite = lw2
-//	if modified {
-//		if err := reload(); err != nil { /* fail */ }
-//		state.lastWrite = lw2
-//	}
-//
-// The caller must hold the lock (for reading or writing).
-func (l *LockFile) ModifiedSince(previous LastWrite) (LastWrite, bool, error) {
-	return LastWrite{}, false, nil
+// TouchedSince indicates if the lock file has been touched since the specified time
+func (l *LockFile) TouchedSince(when time.Time) bool {
+	stat, err := os.Stat(l.file)
+	if err != nil {
+		return true
+	}
+	return when.Before(stat.ModTime())
 }
 
-// Deprecated: Use *LockFile.ModifiedSince.
-func (l *LockFile) Modified() (bool, error) {
-	return false, nil
+func closeHandle(fd uintptr) {
+	windows.Close(windows.Handle(fd))
 }
 
-// Deprecated: Use *LockFile.RecordWrite.
-func (l *LockFile) Touch() error {
-	return nil
+func openHandle(path string, mode int) (uintptr, error) {
+	fd, err := windows.Open(path, mode, windows.S_IWRITE)
+	return uintptr(fd), err
 }
 
-func (l *LockFile) IsReadWrite() bool {
-	return false
+func lockHandle(fd uintptr, lType int16) {
+	ol := new(windows.Overlapped)
+	for windows.LockFileEx(windows.Handle(fd), uint32(lType), reserved, allBytes, allBytes, ol) != nil {
```

Comment:
In the Windows synchronous case (with no FAIL_IMMEDIATELY flag), all errors would be fatal, and retrying indefinitely would result in a hang. Since the API doesn't provide error propagation, I think it's better to panic if this happens. or at least retry a limited number of times.

This is a little bit different than the Unix side of the equation, which needs to loop through signal interruption, although arguably that code should only be doing that when errno == EINTR

Reply:
I think panic makes more sense than retrying an arbitrary number of times. I've pushed the change